lib/classes/Service.js

```diff
@@ -149,16 +149,27 @@ class Service {
         // setup function.name property
         const stageNameForFunction = options.stage || this.provider.stage;
         _.forEach(that.functions, (functionObj, functionName) => {
+          const targetFunction = that.functions[functionName];
           if (!functionObj.events) {
-            that.functions[functionName].events = [];
+            targetFunction.events = [];
```

Comment:
This could be reverted when the translation of the HTTP event is done in the AWS API Gateway plugin.